oak-lucene/src/test/java/org/apache/jackrabbit/oak/composite/CompositeNodeStoreLuceneIndexTest.java

```diff
@@ -0,0 +1,506 @@
+
+        // Add /libs/indexes/luceneTest/@v1 to make this index usable/enabled
+        addOrGetNode(readOnly1Root.getNode("libs"), "indexes", NT_UNSTRUCTURED).addNode(indexName, NT_UNSTRUCTURED).setProperty(versionProp.replace("@", ""), true);
+        readOnlyV1Session.save();
+    }
+
+    void setupIndexAndContentInReadOnlyV2(String indexName, String indexedProperty, int version) throws Exception {
+        String versionProp = (version == 1) ? "@v1" : "@v2";
+        // Add some content to read-only v1 repo
+        for (int i = 0; i < 3; i++) {
+            Node b = readOnly2Root.getNode("libs").addNode("node-" + indexName + "-" + i, NT_UNSTRUCTURED);
+            b.setProperty(indexedProperty, "bar");
+            b.setPrimaryType(NT_UNSTRUCTURED);
+        }
+
+        // index creation
+        createLuceneIndex(readOnlyV2Session, indexName, indexedProperty);
+        readOnly2Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(IndexConstants.USE_IF_EXISTS, "/libs/indexes/" + indexName + "/@" + versionProp);
+        readOnly2Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(FulltextIndexConstants.PROP_REFRESH_DEFN,true);
+
+        // Add /libs/indexes/luceneTest/@v1 to make this index usable/enabled
+        addOrGetNode(readOnly2Root.getNode("libs"), "indexes", NT_UNSTRUCTURED).addNode(indexName, NT_UNSTRUCTURED).setProperty(versionProp.replace("@", ""), true);
+        readOnlyV2Session.save();
+    }
+
+    private String getResult(QueryResult result, String propertyName) throws RepositoryException {
+        StringBuilder buff = new StringBuilder();
+        RowIterator it = result.getRows();
+        while (it.hasNext()) {
+            if (buff.length() > 0) {
+                buff.append(", ");
+            }
+            buff.append(it.nextRow().getValue(propertyName).getString());
+        }
+        return buff.toString();
+    }
+
+    private void createLuceneIndex(JackrabbitSession session, String name, String property) throws Exception {
+
+        Node n = session.getRootNode().getNode(INDEX_DEFINITIONS_NAME);
+        n = n.addNode(name);
+        n.setPrimaryType(IndexConstants.INDEX_DEFINITIONS_NODE_TYPE);
+        n.setProperty("compatVersion", 2);
+        n.setProperty(TYPE_PROPERTY_NAME, "lucene");
+        n.setProperty(REINDEX_PROPERTY_NAME, true);
+        n.setProperty(FulltextIndexConstants.FULL_TEXT_ENABLED, false);
+        n.setProperty("includePropertyNames", new String[]{property});
+        session.save();
+    }
+
+    private Node addOrGetNode(Node parent, String path, String primaryType) throws Exception{
+        return parent.hasNode(path) ? parent.getNode(path) : parent.addNode(path, primaryType);
+    }
+
+    private CompositeRepo restartCompositeRepo(int version) throws Exception{
+        List<MountedNodeStore> nonDefaultStores = Lists.newArrayList();
```

Comment:
int version isn't needed if it's a method of CompositeRepo

Reply:
moved this method to compositeRepo